pkg/controller/nodelifecycle/node_lifecycle_controller.go

```diff
@@ -894,11 +917,20 @@ func (nc *Controller) tryUpdateNodeHealth(node *v1.Node) (time.Duration, v1.Node
 			readyTransitionTimestamp: transitionTime,
 		}
 	}
+	var observedLease *coordv1beta1.Lease
+	if utilfeature.DefaultFeatureGate.Enabled(features.NodeLease) {
+		// Always update the probe time if node lease is renewed.
+		observedLease, _ = nc.leaseLister.Leases(v1.NamespaceNodeLease).Get(node.Name)
+		if observedLease != nil && (!found || savedLease == nil || savedLease.Spec.RenewTime.Before(observedLease.Spec.RenewTime)) {
```

Comment:
nit: Is it necessary to check `found`? It was set 50 lines ago and `savedLease == nil` seems to be sufficient.

Reply:
You are right. We do not need to check `found`. Updated.